lib/web_ui/lib/src/engine/semantics/dialog.dart

```diff
@@ -16,6 +16,39 @@ class Dialog extends PrimaryRoleManager {
     // names its own route an `aria-label` is used instead of `aria-describedby`.
     addFocusManagement();
     addLiveRegion();
+
+    // When a route/dialog shows up it is expected that the screen reader will
+    // focus on something inside it. There could be two possibilities:
+    //
+    // 1. The framework explicitly marked a node inside the dialog as focused
+    //    via the `isFocusable` and `isFocused` flags. In this case, the node
+    //    will request focus directly and there's nothing to do on top of that.
+    // 2. No node inside the route takes focus explicitly. In this case, the
+    //    expectation is to look through all nodes in traversal order and focus
+    //    on the first one.
+    semanticsObject.owner.addOneTimePostUpdateCallback(() {
```

Comment:
how does the debouncing work if there are multiple dialogs created in a update

Reply:
I have not considered it. I hope apps do not create multiple dialogs in a single update. It seems like a messy UI implementation. I think the framework removes semantics from all background routes anyway, so the engine would only observe the top-most one, no?

Reply:
I see. Looking at the code, what will happen right now is that the _last_ dialog to show up will "win" the focus because `addOneTimePostUpdateCallback` callbacks are processed in FIFO order. So whatever order the framework communicates dialogs, the last one to be initialized will win. So I think it should work, but there could be corner-cases. I'm willing to look into this as a follow-up. The current implementation covers all use-cases that we have in the `a11y_assessments` app.